### PR TITLE
dev4.0: CI: Add coverage

### PR DESCRIPTION
**Pull request by wirew0rm**
Add coverage to the CI and upload to codecov.
Meson's documentation is a bit sparse on the topic, so there's still a few things to figure out.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,35 @@ language.settings.xml
 
 # python venv files
 virtualenv/
+
+# files autogenerated by gnuradio
+blocklib/picoscope3000a/meson.build
+blocklib/picoscope3000a/picoscope3000a/meson.build
+blocklib/picoscope4000a/meson.build
+blocklib/picoscope4000a/picoscope4000a/meson.build
```
Comment (frankosterfeld) [inline .gitignore:27]: Need to add picoscope5000a here (or in my PR, whatever is merged afterwards)

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,35 @@ language.settings.xml
 
 # python venv files
 virtualenv/
+
+# files autogenerated by gnuradio
```
Comment (frankosterfeld) [inline .gitignore:23]: these make sense, I would also expect blocklib/digitizers

Comment (wirew0rm) [reply .gitignore:23]: In the long run I would prefer gnuradio/meson/modtool to create them in `./build` for a proper out of tree build. For now I just added everything that git complained about. That means the files in `blocklib/digitizers` where either added to the git by misstake or the autogenerated files had to be manually adjusted and added to git.

**Comment by wirew0rm** [comment]
OK, I'm not sure what I'm doing wrong here, the generated coverage report is empty and it does not honor the excludes correctly. I'm putting this on hold for a while, If anyone can spot the error feel free to improve this, but it's not a priority at the moment.

**Comment by wirew0rm** [comment]
close as it is not applicable to the build system on the current dev4.0 branch anymore.

**Pull request closed without merging**